src/codegen/generators/data_type_generator.rs

```diff
@@ -105,135 +21,153 @@ pub fn generate_data_type_stubs<'a>(llvm: &LLVM<'a>, index: &mut Index<'a>, data
 /// - an alias index entry for sub-range types
 /// - Array type for arrays
 /// - array type for sized Strings
-pub fn generate_data_type<'a>(
-    module: &Module<'a>,
-    llvm: &LLVM<'a>,
-    index: &mut Index<'a>,
-    data_types: &Vec<UserTypeDeclaration>,
-) -> Result<(), CompileError> {
-    for data_type in data_types {
-        match &data_type.data_type {
-            DataType::StructType { name, variables } => {
-                let name = name.as_ref().unwrap();
-                let mut struct_generator = StructGenerator::new(llvm, index);
-                let members: Vec<&Variable> = variables.iter().collect();
-                let (_, initial_value) = struct_generator.generate_struct_type(&members, name)?;
-                index.associate_type_initial_value(name, initial_value);
-                
-            }
-            DataType::EnumType { name: _, elements } => {
-                // generate a global variable for every enum element
-                for (i, element) in elements.iter().enumerate() {
-                    let int_type = llvm.i32_type();
-                    let element_variable = llvm.create_global_variable(
-                        module, 
-                        element,
-                        int_type.as_basic_type_enum(),
-                        Some(int_type.const_int(i as u64, false).as_basic_value_enum()),
-                    );
+pub fn generate_data_types<'ink, 'ind>(llvm: &LLVM<'ink>, index: &'ind Index) -> Result<LLVMTypedIndex<'ink>, CompileError>{
```

Comment:
I'm afraid ind-lifetime has no effect, can be removed

Reply:
Done